crates/matrix-sdk-common/src/linked_chunk/updates.rs

```diff
@@ -294,6 +294,12 @@ impl<Item, Gap> UpdatesInner<Item, Gap> {
         slice
     }
 
+    /// Has the given reader, identified by its [`ReaderToken`], some pending
+    /// updates, or has it consumed all the pending updates?
+    pub(super) fn is_reader_up_to_date(&self, token: ReaderToken) -> bool {
```

Comment:
Can we use [`debug_assertions`](https://doc.rust-lang.org/reference/conditional-compilation.html#r-cfg.debug_assertions) here please?

```rust
#[cfg(debug_assertions)]
```

So that it exists only for non-release build. [That's what `debug_assert!` uses internally](https://github.com/rust-lang/rust/blob/a17780db7b8eebbf42a1cbe6bc9cc83013820ba5/library/core/src/macros/mod.rs#L308).

Reply:
Actually we can't: https://github.com/matrix-org/matrix-rust-sdk/actions/runs/15900921446/job/44843483636?pr=5225

`debug_assert!` uses a `if cfg!` internally, so this means the code below must exist, even if it's not called.

Reply:
Ohhhh, damn…